### PR TITLE
Major update

### PR DESCRIPTION
**Pull request by ebuchman**
Contains lots of updates to bring ethermint up to speed with latest release of go-ethereum (1.5.8) and tendermint (0.8.0).

Some flags are commented out because they were removed from go-ethereum and they need to be explicitly supported elsewhere in the code ... TODO.

Many thanks to @illya13 for getting this to a stable state!

### CODE REVIEW

File: ethereum/backend.go [hunk 1]
```diff
@@ -24,7 +23,6 @@ import (
 // Backend handles the chain database and VM
 type Backend struct {
 	ethereum *eth.Ethereum
-	txSub    *event.TypeMux //Subscription
```
Comment (illya13) [inline ethereum/backend.go]: was using here `*event.TypeMuxSubscription` for `master`

File: ethereum/backend.go [hunk 2]
```diff
@@ -55,10 +53,12 @@ func (s *Backend) APIs() []rpc.API {
 		if v.Namespace == "net" {
 			continue
 		}
-		if txPoolAPI, ok := v.Service.(*eth.PublicTransactionPoolAPI); ok {
```
Comment (illya13) [inline ethereum/backend.go]: `PublicTransactionPoolAPI` has gone, but now `PublicEthereumAPI` has a reference to the same field `txPool`

File: app/app.go [hunk 3]
```diff
@@ -100,31 +101,88 @@ func (app *EthermintApplication) SetOption(key string, value string) (log string
...
+	if err != nil {
+		return err
+	}
+
+	// TODO: only do once a block
+	blockchain := app.backend.Ethereum().BlockChain()
+	currentBlock := blockchain.CurrentBlock()
+	height := currentBlock.Number()
+
+	signer := ethTypes.MakeSigner(blockchain.Config(), height)
```
Comment (illya13) [inline app/app.go]: alt version
```
var signer ethTypes.Signer = ethTypes.FrontierSigner{}
if tx.Protected() {
	signer = ethTypes.NewEIP155Signer(tx.ChainId())
}
```

**Comment by illya13** [comment]
In general - looks good, but I'm not sure is it Ok to avoid using of `txPool` as before

**Comment by illya13** [comment]
glide install: Ok, compiled: Ok, started: Ok
some issues during `tx send`

1) `geth attach ...`
2) `web3.eth.sendTransaction({from: web3.eth.accounts[0], to: "0xb07d8a66361dfab39d8de79682a356dd21d8d8e9", value: web3.toWei(100, "ether")})`
3) some issues in logs

```
[34mINFO[0m[02-10|10:00:24] Executed block                           [34mmodule[0m=state [34mheight[0m=57 [34mvalid txs[0m=0 [34minvalid txs[0m=0I0210 10:00:24.665322 github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core/blockchain.go:1070] imported    1 blocks,     0 txs (  0.000 Mg) in   1.495ms ( 0.000 Mg/s). #33 [ff27afbe…]
[34mINFO[0m[02-10|10:00:24] Recheck txs                              [34mmodule[0m=mempool [34mnumtxs[0m=0
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=2 [34mremoteAddr[0m=127.0.0.1:58746
[34mINFO[0m[02-10|10:00:25] Bad Transaction                          [34mmodule[0m=mempool [34mres[0m="&{CheckTx:code:EncodingError log:\"Encoding error\" }"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"I0210 10:00:25.445875 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] HTTPRestRPC                              [34mmodule[0m=rpcserver [34mmethod[0m=/broadcast_tx_sync [34margs[0m="[<[]uint8 Value>]" [34mreturns[0m="[<core_types.TMResult Value> <error Value>]"
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=11 [34mremoteAddr[0m=127.0.0.1:58747
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=8 [34mremoteAddr[0m=127.0.0.1:58748
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=10 [34mremoteAddr[0m=127.0.0.1:58749I0210 10:00:25.449854 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=7 [34mremoteAddr[0m=127.0.0.1:58750
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=7 [34mremoteAddr[0m=127.0.0.1:58751I0210 10:00:25.449854 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.450357 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.451357 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.452356 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=7 [34mremoteAddr[0m=127.0.0.1:58752
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=6 [34mremoteAddr[0m=127.0.0.1:58754
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=7 [34mremoteAddr[0m=127.0.0.1:58753I0210 10:00:25.468355 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.468355 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=6 [34mremoteAddr[0m=127.0.0.1:58756I0210 10:00:25.468858 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=6 [34mremoteAddr[0m=127.0.0.1:58755
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=5 [34mremoteAddr[0m=127.0.0.1:58757I0210 10:00:25.469854 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=5 [34mremoteAddr[0m=127.0.0.1:58758
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=5 [34mremoteAddr[0m=127.0.0.1:58759
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=4 [34mremoteAddr[0m=127.0.0.1:58760
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=4 [34mremoteAddr[0m=127.0.0.1:58761I0210 10:00:25.470354 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=4 [34mremoteAddr[0m=127.0.0.1:58762I0210 10:00:25.470855 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.474856 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.476356 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Served RPC HTTP response                 [34mmodule[0m=rpcserver [34mmethod[0m=POST [34murl[0m=/broadcast_tx_sync [34mstatus[0m=200 [34mduration[0m=4 [34mremoteAddr[0m=127.0.0.1:58763I0210 10:00:25.476356 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.476855 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.483354 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)I0210 10:00:25.483354 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
[34mINFO[0m[02-10|10:00:25] Timed out                                [34mmodule[0m=consensus [34mdur[0m=888.1476ms [34mheight[0m=58 [34mround[0m=0 [34mstep[0m=RoundStepNewHeight
```

**Comment by illya13** [comment]
also not sure why for a `single tx` we have so many calls to `broadcast`

```
I0210 11:21:03.114876 github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/internal/ethapi/api.go:1076] Tx(0xd7931564cc12e089056e90ec09985c99b327007c675430af05d0ba0cdc6a470f) to: 0xb07d8a66361dfab39d8de79682a356dd21d8d8e9
I0210 11:23:43.113960 github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core/blockchain.go:1070] imported    1 blocks,     0 txs (  0.000 Mg) in        0s (   NaN Mg/s). #12 [9a72b9a1…]
I0210 11:23:43.240245 github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core/blockchain.go:1070] imported    1 blocks,     0 txs (  0.000 Mg) in       1ms ( 0.000 Mg/s). #13 [694b37e0…]
I0210 11:24:34.271199 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:24:57.525439 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:24:59.323926 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:00.742918 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:02.753226 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:05.914147 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:12.808626 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:12.808626 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:15.782027 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:17.373783 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:18.998928 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:21.094942 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:22.572471 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:33.664719 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:33.664719 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:33.664719 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:33.664719 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
I0210 11:25:36.371253 github.com/tendermint/ethermint/ethereum/backend.go:107] Broadcast, err=%!s(<nil>)
```

**Comment by ebuchman** [comment]
I'm sorry, I pushed this prematurely. Thanks for the review and testing.

| replace txPool with validateTx logic

Yeh, I want to avoid the eth txpool, I think, because it's a complicated beast I don't understand. I don't know if it's actually possible to do

File: ethereum/backend.go [hunk 4]
```diff
@@ -99,7 +99,9 @@ func (s *Backend) Config() *eth.Config {
 
 // listen for txs and forward to tendermint
 func (s *Backend) txBroadcastLoop() {
```
Comment (illya13) [inline ethereum/backend.go]: vendor/github.com/ethereum/go-ethereum/eth/handler.go:216

```
func (pm *ProtocolManager) Start() {
	// broadcast transactions
	pm.txSub = pm.eventMux.Subscribe(core.TxPreEvent{})
	go pm.txBroadcastLoop()
...
```
where
```
func (self *ProtocolManager) txBroadcastLoop() {
	// automatically stops if unsubscribe
	for obj := range self.txSub.Chan() {
		event := obj.Data.(core.TxPreEvent)
		self.BroadcastTx(event.Tx.Hash(), event.Tx)
	}
}
```
and `eventMux` is `eth.EventMux()` - so All seems fine here

File: ethereum/backend.go [hunk 5]
```diff
@@ -55,10 +53,12 @@ func (s *Backend) APIs() []rpc.API {
...
 			continue
 		}
-		if txPoolAPI, ok := v.Service.(*eth.PublicTransactionPoolAPI); ok {
+		/*if txPoolAPI, ok := v.Service.(*ethapi.PublicTransactionPoolAPI); ok {
 			s.setFakeTxPool(txPoolAPI)
 			go s.txBroadcastLoop()
-		}
+		}*/
+		// TODO: do we need to overwrite the txPool ?!
+		go s.txBroadcastLoop()
```
Comment (illya13) [inline ethereum/backend.go]: 👎 
This line is inside loop over all APIs
So you do register 8+ subscriptions
Put `if` or move out of loop

Comment (ebuchman) [reply ethereum/backend.go]: Bah! My bad. Thanks

**Comment by illya13** [comment]
👎 
I'm not sure about this line
https://github.com/tendermint/ethermint/blob/b3cd17a68b4f70956646c90cd6b2ac7f1587c390/ethereum/backend.go#L119

Depending on
https://github.com/tendermint/tendermint/blob/764091dfbb035f1b28da4b067526e04c6a849966/rpc/core/routes.go#L28
`broadcast_tx_sync` API, `tx` param: is it `string` or `byte[]`

So for now we do:
byte[] -> hex string -> API call -> handler (hex string as byte[]) -> no hex to byte[] -> decoding error

and by removing `hex.EncodeToString` I'm getting this passed

**Comment by illya13** [comment]
started to have following if I apply 3 fixes described

```
NOTE[02-10|21:50:24] Finalizing commit of block with 1 txs    module=consensus height=56 hash=550861A172D303D7BC95763FA9ECF42B62B971A0 root=0AA52484D2BD2396B44222A8D23F2D14A30144CDEC0A9DF46341AF978D973370
INFO[02-10|21:50:24] Block{
  Header{
    ChainID:        test-chain-Mg4sd5
    Height:         56
    Time:           2017-02-10 21:50:24.797 +0200 EET
    NumTxs:         1
    LastBlockID:    554662FE7E4BC89C21C2A9B67A1A62E5926746A2:1:EC4FA0F95B3C
    LastCommit:     E2BCCCEB41A08C903027629599C69ADAA9A7E48E
    Data:           49B86CA0AAF25808CA66CB126ADFD6E69E85C6DB
    Validators:     FA64FE66447CEA5EB593EEF41B5E0F5ACE4B8AC0
    App:            0AA52484D2BD2396B44222A8D23F2D14A30144CDEC0A9DF46341AF978D973370
  }#550861A172D303D7BC95763FA9ECF42B62B971A0
  Data{
    Tx:[248 110 128 133 4 168 23 200 0 131 1 95 144 148 245 20 228 47 68 239 128 72 159 253 188 103 138 82 36 47 134 241 13 26 137 5 107 199 94 45 99 16 0 0 128 28 160 174 185 72 40 147 80 192 128 32 78 162 176 120 76 77 170 132 208 167 117 39 150 17 242 214 229 10 231 180 58 157 75 160 8 12 99 82 235 20 61 218 55 249 94 98 107 87 195 191 244 53 135 141 212 215 3 7 168 112 71 8 27 141 241 77]
  }#49B86CA0AAF25808CA66CB126ADFD6E69E85C6DB
  Commit{
    BlockID:    554662FE7E4BC89C21C2A9B67A1A62E5926746A2:1:EC4FA0F95B3C
    Precommits: Vote{0:EF61F16B3A5F 55/00/2(Precommit) 554662FE7E4B /03FC03E9FFE7.../}
  }#E2BCCCEB41A08C903027629599C69ADAA9A7E48E
}#550861A172D303D7BC95763FA9ECF42B62B971A0 module=consensus
INFO[02-10|21:50:31] Timed out                                module=consensus dur=3s height=56 round=0 step=RoundStepPropose
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x10 pc=0x500861]

goroutine 51 [running]:
panic(0xdb2060, 0xc0420040b0)
	C:/SOFT/Go/src/runtime/panic.go:500 +0x1af
github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core.NewEVMContext(0x15c0f80, 0xc0423721e0, 0xc0439cb680, 0x15aec20, 0xc042240000, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	e:/projects/go/src/github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core/evm.go:43 +0x1f1
github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core.ApplyTransaction(0xc042012780, 0xc042240000, 0xc04385c400, 0xc043bba3c0, 0xc0439cb680, 0xc0437b2510, 0xc04385c3e0, 0x0, 0x0, 0x0, ...)
	e:/projects/go/src/github.com/tendermint/ethermint/vendor/github.com/ethereum/go-ethereum/core/state_processor.go:99 +0x163
github.com/tendermint/ethermint/app.(*EthermintApplication).DeliverTx(0xc042259d40, 0xc04404ea80, 0x70, 0x70, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	E:/projects/go/src/github.com/tendermint/ethermint/app/app.go:217 +0x5fc
github.com/tendermint/ethermint/vendor/github.com/tendermint/abci/server.(*SocketServer).handleRequest(0xc0421c00a0, 0xc043f21260, 0xc04221e3c0)
	e:/projects/go/src/github.com/tendermint/ethermint/vendor/github.com/tendermint/abci/server/socket_server.go:178 +0x123b
github.com/tendermint/ethermint/vendor/github.com/tendermint/abci/server.(*SocketServer).handleRequests(0xc0421c00a0, 0xc04221e360, 0x15c0e00, 0xc04249e028, 0xc04221e3c0)
	e:/projects/go/src/github.com/tendermint/ethermint/vendor/github.com/tendermint/abci/server/socket_server.go:159 +0x164
created by github.com/tendermint/ethermint/vendor/github.com/tendermint/abci/server.(*SocketServer).acceptConnectionsRoutine
	e:/projects/go/src/github.com/tendermint/ethermint/vendor/github.com/tendermint/abci/server/socket_server.go:114 +0x225
```

**Comment by illya13** [comment]
'app.Query is temporarily disabled as it needs to be updated for the new go-ethereum rpc Client.'
this is not true any more

**Pull request merged**